google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddAffiliateLocationExtensions.java

```diff
@@ -0,0 +1,397 @@
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      gae.printStackTrace();
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, long campaignId, long chainId) {
+    String feedResourceName =
+        createAffiliateLocationExtensionFeed(googleAdsClient, customerId, chainId);
+    // After the completion of the feed creation operation above the added feed will not
+    // be available for usage in a campaign feed until the feed mappings are created.
+    // We will wait with an exponential back-off policy until the feed mappings have
+    // been created.
+    FeedMapping feedMapping = waitForFeedToBeReady(googleAdsClient, customerId, feedResourceName);
+    createCampaignFeed(
+        googleAdsClient, customerId, campaignId, feedMapping, feedResourceName, chainId);
+  }
+
+  /** Creates an affiliate location extension feed. */
+  private String createAffiliateLocationExtensionFeed(
+      GoogleAdsClient googleAdsClient, long customerId, long chainId) {
+    // Deletes all existing location extension feeds. This is an optional step,
+    // but is required for this code example to run correctly more than once. This is because:
+    //   1. Google Ads only allows one location extension feed per email address.
+    //   2. A Google Ads account cannot have a location extension feed and an affiliate
+    // location extension feed at the same time.
+    deleteLocationExtensionFeeds(googleAdsClient, customerId);
+
+    // Creates a feed that will sync to retail addresses for a given retail chain ID.
+    // Do not add FeedAttributes to this object as Google Ads will add
+    // them automatically because this will be a system generated feed.
+    Feed feed =
+        Feed.newBuilder()
+            .setName("Affiliate Location Extension feed #" + System.currentTimeMillis())
+            .setAffiliateLocationFeedData(
+                AffiliateLocationFeedData.newBuilder()
+                    .addChainIds(chainId)
+                    .setRelationshipType(AffiliateLocationFeedRelationshipType.GENERAL_RETAILER))
+            // Since this feed's contents will be managed by Google,
+            // you must set its origin to GOOGLE.
+            .setOrigin(FeedOrigin.GOOGLE)
+            .build();
+
+    // Constructs an operation to create the feed.
+    FeedOperation op = FeedOperation.newBuilder().setCreate(feed).build();
+
+    // Connects to the API.
```

Comment:
Nit: change to "Creates the FeedServiceClient" (consistent with other examples).